app/cronjob/trailingTrade.js

```diff
@@ -25,11 +25,19 @@ const {
   saveDataToCache
 } = require('./trailingTrade/steps');
 const { errorHandlerWrapper } = require('../error-handler');
+const { cache } = require('../helpers');
 
 const execute = async (rawLogger, symbol) => {
   const logger = rawLogger.child({ jobName: 'trailingTrade' });
 
   await errorHandlerWrapper(logger, 'Trailing Trade', async () => {
+    if ((await cache.hget(`execute-trailing-trade`, symbol)) === 'true') {
+      // do nothing, there is another execution task running
+      return;
```

Comment:
You can set a flag in cache here to mark there was an attempt to execute processing and check for that flag later to reexecute processing again.

Reply:
Sorry, I resolved it accidentally.
It looks like @uhliksk suggesting the queue?

Reply:
@chrisleekr 
Yes, @uhliksk is suggesting use a queue instead of simply do nothing.
While I am worried about the delayed execution because of the queue.
To be honest, I don't know which solution is most suitable.

If you have a better idea, drop it here and I will work on it :)

Reply:
@habibalkhabbaz 

Since we have changed the bot to use WebSocket, this kind of issue is anticipated. Although, I would expect the tick will happen all the time. But the reality was some coins are not very active. 

One of the best Redis queue tools I've used is called Bull queue - https://github.com/OptimalBits/bull
Easy to implement, there is UI as well. 

Since you've added the code to prevent multiple executions, adding a queue can be adding extra stability.
The logic would be like,

1. Check if the symbol is current in the execution - https://github.com/chrisleekr/binance-trading-bot/pull/448/files/1bd376ac271322d07df190e5ed8ec2b4c9d6f97d..d5a41033b5d5224386bd802a6a4bfd06be3b1b59#
2. If true, then queue to Bull. The Bull has a mechanism that can prevent duplicate jobs by setting ID. 

Now I am thinking, it would be much better if adding the execution job to Bull whenever receiving the tick. 
In addition, all those manual executions we added when the manual task is initiated from the UI can be added to the queue as well. 

Although it would require some refactoring...

Reply:
@chrisleekr 

Thanks for the inspiration!
In fact, at the beginning of refactoring the code to use the WebSocket, I use the queue but I thought it is not necessary and nothing critical will happen if I remove it and I removed it.
However, after I read your comment I started to rethink and I found it's better to use it, especially when we trigger manual actions as you said. 
It can improve stability and management of actions.

I can work on the changes.

Reply:
@habibalkhabbaz 
That is awesome. 
Sorry for not being able to help you code. Family first...

Reply:
@chrisleekr 
Nothing wrong. Enjoy with your family. We will do what we can do with this bot :)

Reply:
Thanks for understanding. :)

Reply:
@habibalkhabbaz 

I was thinking, if you are going to implement the queue, let's merge this PR and make a new PR for the queue.
I think your code is stable.

What do you think?

Reply:
Hello @chrisleekr 
Yes, I see it's fine, we can merge now and make a new PR later for the queue.